pkg/snapstore/generic_s3_snapstore.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snapstore
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+// s3AuthOptions contains all needed options to authenticate against a S3-compatible store.
+type s3AuthOptions struct {
+	endpoint           string
+	region             string
+	disableSSL         bool
+	insecureSkipVerify bool
+	accessKeyID        string
+	secretAccessKey    string
```

Comment:
need an `enableEncrytion` option?

Reply:
No, this has been removed, see gardener/gardener-extension-provider-kubevirt#59. Encryption is not supported by OCS, correct? It only made sense in the context of a "generic" S3-compatible store, but this is gone now.